contributed/DC2_object_deredden.ipynb

```diff
@@ -0,0 +1,369 @@
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Looking up the ebv value at each position is now a simple procedure with `dustmaps`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sfd = SFDQuery()\n",
+    "ebvvec = sfd(coords)\n",
+    "df['ebv'] = ebvvec"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To de-redden the magnitudes, we simply need to subtract of A_lambda/E(B-V)*E(B-V) for each band:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for i,band in enumerate(['u','g','r','i','z','y']):\n",
+    "    df[f'mag_{band}_cModel_dered']= df[f'mag_{band}_cModel']-df['ebv']*band_a_ebv[i]\n",
+    "    df[f'mag_{band}_dered'] = df[f'mag_{band}']-df['ebv']*band_a_ebv[i]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "That's it!  But, to check if our dereddening worked correctly we'll make a few plots.  Let's see what our E(B-V) map looks like in this tract:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig = plt.figure(figsize=(15,12))\n",
+    "plt.scatter(df['ra'][::10],df['dec'][::10],s=15,c=df['ebv'][::10],cmap='hot')\n",
+    "plt.xlabel(\"RA (degrees)\",fontsize=18)\n",
+    "plt.ylabel(\"DEC (degrees)\",fontsize=18)\n",
+    "plt.colorbar();"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We see varying amounts of foreground dust with an E(B-V) going as high as 0.05.  For the u-band this means a de-reddening as high as 0.25 magnitudes, with lesser effects for logner wavelength bands.  Let's Select some non-blended, non-extended samples from the region with high E(B-V), and compare to a tabulated list of some of the star colors used in IMSIM: the Kurucz and \"old mlt\" (red) stars:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mask = (df['ebv']>.04) & (df['blendedness']<.05) & (df['extendedness']<.1) & (df['mag_i_cModel_dered']<23.5)\n",
+    "gooddf = df[mask]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For comparison, we have queried the truth fluxes with and without Milky Way dust from the postgres SQL database (which you can learn about in the postgres_truth.ipynb notebook in the tutorials directory).  We ran a query to grab true fluxes for stars in the same region of sky as the example tract.  To save time we've saved this data in a parquet file that we will read in with Pandas:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "stardf = pd.read_parquet(\"data/demo_star_fluxes_colors_dust.parquet\",engine='pyarrow')"
```

Comment:
Should we check in the script or notebook that was used to generate this file?

Reply:
I changed the notebook to read in from GCRCatalogs star truth table, and removed the dependency on this file.